src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  const stateClone = { ...state };
+
+  for (const action of actions) {
+    const { extraData, type, keysToRemove } = action;
+
+    switch (type) {
+      case 'clear':
+        for (const key in stateClone) {
+          delete stateClone[key];
+        }
+        break;
+
+      case 'addProperties':
+        Object.assign(stateClone, extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          delete stateClone[key];
+        }
+        break;
+
+      default:
+        throw new Error(`Wrong type`);
```

Comment:
[CODE STYLE]: The `switch` statement is used correctly according to the checklist. However, consider moving the line `states.push({ ...stateClone });` outside of the `switch` block to follow the [DONT REPEAT YOURSELF] principle. This way, you avoid repeating the cloning operation in each case and ensure it's done once per iteration after the `switch` statement.